### ISSUE TITLE
[Bug]: DialogService.ShowEx results in Microsoft.JSInterop.JSDisconnectedException after page refresh (as of 1.7.64)

### ISSUE THREAD

**Issue by jaypalexa** [fgilde/MudBlazor.Extensions#32]
### What happened?

Hello, Florian -- Since the 1.7.64 release, we are now seeing a `Microsoft.JSInterop.JSDisconnectedException` message in the browser console when we call `DialogService.ShowEx` after a hard page refresh.

```
var options = new DialogOptionsEx { CloseOnEscapeKey = true, DragMode = MudDialogDragMode.Simple, CloseButton = true };
await DialogService.ShowEx<TestDialog>("Test Dialog", options);
```

(Admittedly this sounds like a strange scenario.  We do these hard page refreshes when we are editing code / hot module reloading.)

Suspect it has something to do with recent changes around `InitializeMudBlazorExtensionsAsync()`...the call to which we did _not_ remove, even though it is marked as obsolete.

If we do remove the `InitializeMudBlazorExtensionsAsync()` then it appears we have to pass a `JsRuntime` reference as part of `DialogOptionsEx` (which we actually tried, but it still resulted in this exception).

```
Error: Microsoft.JSInterop.JSDisconnectedException: JavaScript interop calls cannot be issued at this time. This is because the circuit has disconnected and is being disposed.
   at Microsoft.AspNetCore.Components.Server.Circuits.RemoteJSRuntime.BeginInvokeJS(Int64 asyncHandle, String identifier, String argsJson, JSCallResultType resultType, Int64 targetInstanceId)
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, CancellationToken cancellationToken, Object[] args)
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at MudBlazor.Extensions.Helper.JsRuntimeExtensions.IsNamespaceAvailableAsync(IJSRuntime js, String ns)
   at MudBlazor.Extensions.Helper.JsImportHelper.InitializeMudBlazorExtensionsCoreAsync(IJSRuntime runtime, Boolean force)
   at MudBlazor.Extensions.Helper.JsImportHelper.GetInitializedJsRuntime(Object field, IJSRuntime fallback)
   at MudBlazor.Extensions.DialogServiceExt.InjectOptionsAsync(IDialogReference dialogReference, DialogOptionsEx options)
   at MudBlazor.Extensions.DialogServiceExt.InjectOptionsAsync(Task`1 dialogReference, DialogOptionsEx options)
   at MudBlazor.Extensions.DialogServiceExt.ShowAndInject(IDialogService dialogService, Type type, String title, DialogOptionsEx options, DialogParameters parameters)
   at MudBlazor.Extensions.Core.MudExDialogReferenceExtensions.AsMudExDialogReferenceAsync[TComponent](Task`1 dialogReference)
   at TestApp.Index.OnOpenTestDialogClick() in C:\temp\TestApp\Index.razor.cs:line 15
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at MudBlazor.MudBaseButton.OnClickHandler(MouseEventArgs ev)
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

We are going to pin to 1.7.63 for the time being.

I will try to create a minimal sample application and upload it here soon.

Thanks for all your work on this project.  It is much appreciated.

### Expected Behavior

No exception

### What browser are you using?

Chrome

### Sample Solution

[TestApp.zip](https://github.com/fgilde/MudBlazor.Extensions/files/12226464/TestApp.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jaypalexa** [comment]
Oh, no.  The culprit seems to be a conflict with CodeBeam.MudBlazor.Extensions.  I have attached a .zip file containing a test application.

[TestApp.zip](https://github.com/fgilde/MudBlazor.Extensions/files/12217270/TestApp.zip)

TestApp.csproj
```
<PackageReference Include="CodeBeam.MudBlazor.Extensions" Version="6.5.6" />
```

Program.cs
```
using MudBlazor.Extensions;
using MudBlazor.Services;
using MudExtensions.Services; // <-- hooking in CodeBeam.MudBlazor.Extensions causes the exception

var builder = WebApplication.CreateBuilder(args);
builder.Services.AddHttpContextAccessor();
builder.Services.AddRazorPages();
builder.Services.AddServerSideBlazor();
builder.Services.AddMudServices();
builder.Services.AddMudExtensions();

var app = builder.Build();
app.UseDeveloperExceptionPage();
app.UseHttpsRedirection();
app.UseStaticFiles();
app.UseRouting();
app.MapBlazorHub();
app.MapFallbackToPage("/_Host");

app.Run();
```

**Comment by fgilde** [comment]
Problem is you didnt have called AddMudExtensions(); from my package. Becuase you have added using of CodeBeam you called the codebeam AddMudExtensions(); (Same name but different method). However to avoid this you have 2 options. You can call it without extension invocation like this 

`MudBlazor.Extensions.ServiceCollectionExtensions.AddMudExtensions(builder.Services); `

Or you just replace the line `builder.Services.AddMudServices();`  with `builder.Services.AddMudServicesWithExtensions();`

**Comment by jaypalexa** [comment]
Thank you!  Yes, both of those options appear to work:

Option 1:
```
using MudBlazor.Extensions;
using MudBlazor.Services;
using MudExtensions.Services;

// this works:
builder.Services.AddMudServices();
builder.Services.AddMudExtensions();
MudBlazor.Extensions.ServiceCollectionExtensions.AddMudExtensions(builder.Services);
```

Option 2:
```
using MudBlazor.Extensions;
//using MudBlazor.Services; // <-- not needed
using MudExtensions.Services;

// this works:
builder.Services.AddMudExtensions();
builder.Services.AddMudServicesWithExtensions();
```

I thought we were running into an issue with Option 2 due to having a Codebeam `MudSelectExtended` component on the dialog, but it was because we had not included `AddMudExtensions()` as well as `AddMudServicesWithExtensions`.

(Honestly, we would prefer to use your `MudExChipSelect` component instead (and then we could get rid of Codebeam completely to minimize these kinds of issues), but we have run into problems being able to consistently be able to get focus into the Filter textbox...perhaps I will open another issue for that, if you are interested...I have updated the TestApp.zip to have dialogs with both components.)

[TestApp.zip](https://github.com/fgilde/MudBlazor.Extensions/files/12226859/TestApp.zip)

Off to buy you a well-deserved cup of coffee.  I very much appreciate all of your work and attention.  All the best.

**Comment by fgilde** [comment]
I know about the focus bug, but it was a bug from MudBlazor and resolved in 6.5.0 already. 
But I will have a look.

**Comment by jaypalexa** [comment]
Yes, sorry, of course.  Closing this issue.
